### PR TITLE
ref(performance): Remove unsupported transaction summary features

### PR DESCRIPTION
**Pull request by 0Calories**
As we move to a spans-only architecture, a lot of these features will no longer be supported. For now, we're hiding them behind a feature flag as we transition away from them.

For users that have this flag enabled, the following will be hidden:

- Spans tab
- Suspect spans table
- Get samples button

### CODE REVIEW

File: static/app/views/performance/transactionSummary/transactionOverview/content.tsx [hunk 1]
```diff
@@ -332,15 +332,22 @@ function SummaryContent({
...
   return (
     <Fragment>
       <Layout.Main>
-        <FilterActions>
-          <Filter
-            organization={organization}
-            currentFilter={spanOperationBreakdownFilter}
-            onChangeFilter={onChangeFilter}
-          />
+        <FilterActions hasTransactionSummaryCleanupFlag>
```
Comment (narsaynorath) [inline static/app/views/performance/transactionSummary/transactionOverview/content.tsx:342]:
```suggestion
        <FilterActions hasTransactionSummaryCleanupFlag={hasTransactionSummaryCleanupFlag}>
```

**Review by narsaynorath — APPROVED** [review]
Just one comment on passing a flag down to the styled component

File: static/app/views/performance/transactionSummary/transactionOverview/content.tsx (hunk 1 above)
Comment (0Calories) [reply static/app/views/performance/transactionSummary/transactionOverview/content.tsx:342]: hmmm I'm of the opinion that passing booleans as props is a lot cleaner if you just attach the prop like I have here

Comment (0Calories) [reply static/app/views/performance/transactionSummary/transactionOverview/content.tsx:342]: but unsure whether this pattern is disliked or not by others

Comment (0Calories) [reply static/app/views/performance/transactionSummary/transactionOverview/content.tsx:342]: nevermind good catch!

**Pull request merged**